packages/indicators/src/Builder/Builder.ts

```diff
@@ -23,29 +20,24 @@ export abstract class Builder {
    */
   abstract getElementCodes(): string[];
 
-  abstract getAggregations(): Aggregation[];
+  abstract getAggregationListsMap(): AggregationListsMap;
 
+  /**
+   * @param extraAggregations Useful for passing aggregations top-down to nested indicators
+   */
   buildAnalytics = (
-    populatedAnalyticsRepo: AnalyticsRepository,
+    analyticsRepo: AnalyticsRepository,
     buildersByIndicator: Record<string, Builder>,
-    fetchOptions: FetchOptions,
-  ): Analytic[] => {
-    if (!populatedAnalyticsRepo.isPopulated()) {
-      throw new Error(
-        'buildAnalytics expects that the provided analyticsRepository is already populated',
-      );
-    }
-
-    return this.buildAnalyticValues(
-      populatedAnalyticsRepo,
-      buildersByIndicator,
-      fetchOptions,
-    ).map(value => ({ ...value, dataElement: this.indicator.code }));
-  };
+    extraAggregations: Aggregation[] = [],
```

Comment:
Those aggregations are passed top-down between nested indicators, similar to how `props` are propagated in React components.

I couldn't of a better name, but please let me know if you do!

Reply:
I like it!